### ISSUE TITLE
Fig suggests folder paths while searching inside `vim` editor

### ISSUE THREAD

**Issue by mschrage** [withfig/fig#178]
> but while i'm searching in vim editor, i don't want to autocomplete folder paths

**Comment by roomedia** [comment]
I'm using iTerm2.
It starts with 3 Split panes using Window Arrangement.
When I enter `vim` with `vi test.cpp` (or any other filename, existing file) and press `/` to search something, `fig` autocomplete folder path.

new window -> ok
new pane -> fine

MacBook Pro (Retina, 13-inch, Early 2015, 2.7GHz Dual Core Intel Core i5, Big Sur 11.1)
iTerm (Build 3.4.8)
fig (1.0.42)

```
> fig settings
{
  "autocomplete.addStatusToTerminalTitle" : false,
  "autocomplete.disable" : false,
  "developer.debugMode" : false
}
```

https://user-images.githubusercontent.com/7322527/122001917-02547f00-cdec-11eb-90f3-d768e4e4d395.mov

**Comment by mattschrage** [comment]
Can you take a screenshot of the debugger menu when this happens? (You can find it by clicking the Fig icon in the menu bar)

**Comment by roomedia** [comment]
![스크린샷 2021-06-22 오후 2 21 29](https://user-images.githubusercontent.com/7322527/122867996-52d85900-d365-11eb-965e-b2de9e692e42.png)
![스크린샷 2021-06-22 오후 2 21 46](https://user-images.githubusercontent.com/7322527/122868048-61267500-d365-11eb-98f4-e8374bd5bd56.png)
![스크린샷 2021-06-22 오후 2 20 58](https://user-images.githubusercontent.com/7322527/122868101-73081800-d365-11eb-990d-ae3c67b2935d.png)

**Comment by mattschrage** [comment]
Thanks so much for sharing! This definitely shouldn't happen (in fact, I'm struggling to reproduce it on my machine).

<img width="859" alt="Screen Shot 2021-06-21 at 11 26 43 PM" src="https://user-images.githubusercontent.com/3155109/122874876-6396b980-d2e8-11eb-87ba-a2f14803449d.png">

This looks like it might be an issue with the shell integration. Can you check if `~/.fig/fig.sh` is sourced in your `.bashrc` and `.bash_profile`?

I'd love to jump on a call and help debug :)

**Comment by roomedia** [comment]
here're my `.bashrc` and `.bash_profile`
![스크린샷 2021-06-22 오후 4 59 56](https://user-images.githubusercontent.com/7322527/122886838-46f79180-d37b-11eb-9fa4-adbea4797226.png)

I think I find a way to reproduce this bug.
1. launch `iTerm 2`
2. split panes using `cmd` + `d` or `cmd` + `shift` + `d`
3. save window arrangement using `cmd` + `shift` + `s`
4. set that arrangement as default in `Preferences` - `Arrangements` - `Set Default`
5. relaunch `iTerm 2`
6. `vi example.run`, type `/`

https://user-images.githubusercontent.com/7322527/122893191-0f8be380-d381-11eb-9152-3c7a447c52d5.mp4

**Comment by mattschrage** [comment]
Thanks for outlining these steps! Very helpful. I'll investigate this further.

In the meantime, I've noticed you're on an old build (v1.0.42). Try updating to the latest version and see if that helps at all. We've refactored our iTerm integration, so that might have been the issue originally.

**Comment by roomedia** [comment]
v1.0.45 still has the problem. 🥲

**Comment by ohjann** [comment]
Bumping this because also experiencing the same issue on fig `v1.0.47`: iTerm2 `3.4.8`, neovim `0.4.4`, zsh `5.8`, macos Big Sur `11.5.2`

**Comment by mschrage** [comment]
Hey @ohjann! Thanks for bumping.  This should be resolved in the next release (v1.0.49). We've changed how we determine 'edit buffer'.

Before we had to listen to raw keystrokes and so we could run into weird issues like this, now we get the edit buffer directly from the shell!
